src/core/sys/dragonflybsd/sys/link_elf.d

```diff
@@ -0,0 +1,81 @@
+/**
+ * D header file for DragonFlyBSD.
+ *
+ * $(LINK2 http://svnweb.freebsd.org/base/head/sys/sys/link_elf.h?view=markup, sys/link_elf.h)
+ * Authors: Diederik de Groot(port:DragonFlyBSD)
+ * Copied:  From core/sys/freebsd/sys
+ */
+module core.sys.dragonflybsd.sys.link_elf;
+
+version (DragonFlyBSD):
+
+extern (C) nothrow @system:
+
+import core.stdc.stdint : uint64_t;
+import core.sys.dragonflybsd.sys.elf;
+
+version(D_LP64)
+    enum __ELF_NATIVE_CLASS = 64;
+else
+    enum __ELF_NATIVE_CLASS = 32;
+
+template ElfW(string type)
+{
+    mixin("alias Elf"~__ELF_NATIVE_CLASS.stringof~"_"~type~" ElfW;");
+}
+
+enum LA_SER_ORIG =      0x01;
+enum LA_SER_LIBPATH =   0x02;
+enum LA_SER_RUNPATH =   0x04;
+enum LA_SER_CONFIG =    0x08;
+enum LA_SER_DEFAULT =   0x40;
+enum LA_SER_SECURE =    0x80;
+
+struct link_map
+{
+    char*           l_addr;
+
+    version (MIPS32)
+        char*       l_offs;
+    version (MIPS64)
+        char*       l_offs;
+
+    char*           l_name;
+    void*           l_ld;
+    link_map*       l_next, l_prev;
+}
+alias link_map Link_map;
+
+enum
+{
+    RT_CONSISTENT,
+    RT_ADD,
+    RT_DELETE,
+}
+
+struct r_debug
+{
+    int             r_version;
+    link_map*       r_map;
+    void function(r_debug*, link_map*) r_brk;
+};
+
+struct dl_phdr_info
+{
+    ElfW!"Addr"     dlpi_addr;
+    char*           dlpi_name;
+    ElfW!"Phdr"*    dlpi_phdr;
+    ElfW!"Half"     dlpi_phnum;
+    uint64_t        dlpi_adds;
+    uint64_t        dlpi_subs;
+    size_t          dlpi_tls_modid;
+    void*           dlpi_tls_data;
+};
+
+
+private alias int function(dl_phdr_info*, size_t, void *) dl_iterate_phdr_cb;
```

Comment:
When I asked about the `extern(C)` here, I honestly didn't know if it was required.  You said the tests didn't need it, but that's understandable [since you don't call the function it's used from below](https://github.com/dlang/druntime/pull/1999/files#diff-f85f6c9c8c517f0652109bd5b27466a5R814).  Thinking about it, I guess since it's a type for an `extern(C)` function, which don't mangle their types anyway, I suppose the `extern(C)` adds nothing.  I tested it on linux by removing this attribute there and the tests ran fine.

If anyone knows why it might be needed, comment here.

Reply:
Should I remove the second 'extern', ie:
```
extern (C) nothrow @system:
...
extern int dl_iterate_phdr(dl_iterate_phdr_cb __callback, void*__data);
extern int ...
```
should become:
```
extern (C) nothrow @system:
...
int dl_iterate_phdr(dl_iterate_phdr_cb __callback, void*__data);
int ...
```
Is that what you want to tell me ?

Reply:
No, I was talking about [the `extern(C)` you already removed above](https://github.com/dlang/druntime/pull/2002#discussion_r161364963).  I don't think you need to change it again, but I'm not sure, so I left it open if anybody else wanted to comment here.

Reply:
Removing the extern(C) at the top of link_elf.d is also possible, but i would have to add notrown to the extern's at the bottom. So this is almost more of a style issue.

Reply:
It has nothing to do with style, these attributes normally affect linkage and functionality. However, I think they don't do anything in these cases so it's good you removed them from the bottom.  The one up top is needed though.

Reply:
Ok Reverting the last change. Thanks for the update and review !